src/metpy/calc/basic.py

```diff
@@ -332,6 +332,9 @@ def heat_index(temperature, relative_humidity, mask_undefined=True):
                    * (units.Quantity(87., 'delta_degF') - delta[sel]))
         hi[sel] = hi[sel] + rh85adj
 
+    # Convert heat index to temperature units
```

Comment:
What about instead of adding the conversion here, just modify the return so that at the end of the function it's just:
```python
return hi.to(temperature.units)
```
?

Since the data are calculated in Fahrenheit this avoids any unit conversion under the `mask_undefined` block.

Reply:
No problem - fixed in latest commit

Reply:
Thanks!